homeassistant/components/nut/config_flow.py

```diff
@@ -0,0 +1,143 @@
+"""Config flow for Network UPS Tools (NUT) integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_ALIAS,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_RESOURCES,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from . import PyNUTData, find_resources_in_config_entry, pynutdata_status
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, SENSOR_TYPES
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SENSOR_DICT = {sensor_id: SENSOR_TYPES[sensor_id][0] for sensor_id in SENSOR_TYPES}
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_RESOURCES): cv.multi_select(SENSOR_DICT),
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_ALIAS): str,
```

Comment:
> If there are multiple UPSs attached to the NUT server, enter the name UPS to query in the 'Alias' field.

Can we make this a dropdown as a 2nd step ? Or even better, a multi select list using `cv.multi_select` to allow the user to pick the ones to integrate.

Reply:
Could do something like this in a followup:

Step 1 Basic Credentials:
```
CONF_HOST
CONF_PORT
CONF_USERNAME
CONF_PASSWORD
```

Step 2 Choose Device and Resources(at this point we know which resources the device supports so we could filter down the list even more)
```
CONF_ALIAS (hidden if only one)
CONF_RESOURCES
```

Reply:
Saved in followup notes

Reply:
Done here
https://github.com/home-assistant/core/pull/33803